extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/HealthOASFilter.java

```diff
@@ -0,0 +1,217 @@
+package io.quarkus.smallrye.openapi.deployment;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.openapi.OASFilter;
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.eclipse.microprofile.openapi.models.Operation;
+import org.eclipse.microprofile.openapi.models.PathItem;
+import org.eclipse.microprofile.openapi.models.Paths;
+import org.eclipse.microprofile.openapi.models.media.Content;
+import org.eclipse.microprofile.openapi.models.media.MediaType;
+import org.eclipse.microprofile.openapi.models.media.Schema;
+import org.eclipse.microprofile.openapi.models.responses.APIResponse;
+import org.eclipse.microprofile.openapi.models.responses.APIResponses;
+
+import io.smallrye.openapi.api.models.ComponentsImpl;
+import io.smallrye.openapi.api.models.OperationImpl;
+import io.smallrye.openapi.api.models.PathItemImpl;
+import io.smallrye.openapi.api.models.PathsImpl;
+import io.smallrye.openapi.api.models.media.ContentImpl;
+import io.smallrye.openapi.api.models.media.MediaTypeImpl;
+import io.smallrye.openapi.api.models.media.SchemaImpl;
+import io.smallrye.openapi.api.models.responses.APIResponseImpl;
+import io.smallrye.openapi.api.models.responses.APIResponsesImpl;
+
+/**
+ * This adds /health endpoint to the document.
+ */
+public class HealthOASFilter implements OASFilter {
+    private static final String ROOT_PATH = "quarkus.smallrye-health.root-path";
+    private static final String LIVENESS_PATH = "quarkus.smallrye-health.liveness-path";
+    private static final String READINESS_PATH = "quarkus.smallrye-health.readiness-path";
```

Comment:
Suggestion: maybe we can retrieve the values represented by these configuration through the config root classes:
 https://github.com/quarkusio/quarkus/blob/e76d50703dbe5e116a3ce9ce7b57e65c2ca8f017/extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthConfig.java#L8

We might need to create a spi module for that.